test/benchmarks/benchmark_slidehash.cc

```diff
@@ -0,0 +1,85 @@
+#include <stdint.h>
+#include <stdint.h>
+#include <limits.h>
+
+#include "zbuild.h"
+
+extern "C" {
+#  include "zbuild.h"
+#  include "zutil.h"
+#  include "cpu_features.h"
+#  include "deflate.h"
+}
+
+#include <benchmark/benchmark.h>
+
+#define MAX_RANDOM_INTS 32768
+
+typedef void (*slide_hash_func)(deflate_state *s);
+
+class slide_hash: public benchmark::Fixture {
+private:
+    uint16_t *l0;
+    uint16_t *l1;
+    deflate_state *s_g;
+
+public:
+    void SetUp(const ::benchmark::State& state) {
+        l0 = (uint16_t *)malloc(HASH_SIZE * sizeof(uint16_t));
+
+        for (int32_t i = 0; i < HASH_SIZE; i++) {
+            l0[i] = rand();
+        }
+
+        l1 = (uint16_t *)malloc(MAX_RANDOM_INTS * sizeof(uint16_t));
+
+        for (int32_t i = 0; i < MAX_RANDOM_INTS; i++) {
+            l1[i] = rand();
+        }
+
+        deflate_state *s = (deflate_state*)malloc(sizeof(deflate_state));
+        s->head = l0;
+        s->prev = l1;
+        s_g = s;
+    }
+
+    void Bench(benchmark::State& state, slide_hash_func slide_hash) {
+        s_g->w_size = (uint32_t)state.range(0);
+
+        for (auto _ : state) {
+            slide_hash(s_g);
+            benchmark::DoNotOptimize(s_g);
+        }
+    }
+
+    void TearDown(const ::benchmark::State& state) {
+        free(l0);
+        free(l1);
+    }
+};
+
+
+BENCHMARK_DEFINE_F(slide_hash, c)(benchmark::State& state) {
+    Bench(state, slide_hash_c);
+}
+BENCHMARK_REGISTER_F(slide_hash, c)->RangeMultiplier(2)->Range(1024, MAX_RANDOM_INTS);
+#ifdef X86_SSE2_SLIDEHASH
+BENCHMARK_DEFINE_F(slide_hash, sse2)(benchmark::State& state) {
+    if (!x86_cpu_has_sse2) {
+        state.SkipWithError("CPU does not support SSE2");
+        return;
+    }
+    Bench(state, slide_hash_sse2);
+}
+BENCHMARK_REGISTER_F(slide_hash, sse2)->RangeMultiplier(2)->Range(1024, MAX_RANDOM_INTS);
+#endif
+#ifdef X86_AVX2_SLIDEHASH
```

Comment:
Probably _should_ be fixed in CmakeLists.txt and configure but, this needs to be:
`X86_AVX2`

Reply:
Yes I just noticed that as well.

Reply:
Fixed.